packages/ai-jsx/src/lib/anthropic.tsx

```diff
@@ -0,0 +1,178 @@
+ * @param children The children to render.
+ * @param chatModel The chat model to use.
+ * @param completionModel The completion model to use.
+ * @param client The OpenAI client.
+ */
+export function OpenAI({
+  children,
+  chatModel,
+  completionModel,
+  client,
+  ...defaults
+}: { children: Node; client?: Anthropic } & AnthropicModelChoices & ModelProps) {
+  let result = children;
+
+  if (client) {
+    result = <anthropicClientContext.Provider value={client}>{children}</anthropicClientContext.Provider>;
+  }
+
+  if (chatModel) {
+    result = (
+      <ChatProvider component={AnthropicChatModel} {...defaults} model={chatModel}>
+        {result}
+      </ChatProvider>
+    );
+  }
+
+  // TS is correct that this should never happen, but we'll check for it anyway.
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (completionModel) {
+    throw new AIJSXError(
+      'Completion models are not supported by Anthropic',
+      ErrorCode.AnthropicDoesNotSupportCompletionModels,
+      'user'
+    );
+  }
+
+  return result;
+}
+
+interface AnthropicChatModelProps extends ModelPropsWithChildren {
+  model: ValidChatModel;
+}
+export async function* AnthropicChatModel(
+  props: AnthropicChatModelProps,
+  { render, getContext, logger }: AI.ComponentContext
+): AI.RenderableStream {
+  const messageElements = await render(props.children, {
+    stop: (e) =>
+      e.tag == SystemMessage ||
+      e.tag == UserMessage ||
+      e.tag == AssistantMessage ||
+      e.tag == FunctionCall ||
+      e.tag == FunctionResponse,
+  });
+  yield AI.AppendOnlyStream;
+  const messages = await Promise.all(
+    messageElements.filter(AI.isElement).map(async (message) => {
+      switch (message.tag) {
+        case UserMessage:
+          return `${Anthropic.HUMAN_PROMPT}: ${await render(message)}`;
+        case AssistantMessage:
+          return `${Anthropic.AI_PROMPT}: ${await render(message)}`;
+        case SystemMessage:
+          throw new AIJSXError(
+            'Anthropic models do not support SystemMessage. Change your user message to instruct the model what to do.',
+            ErrorCode.AnthropicDoesNotSupportSystemMessage,
+            'user'
+          );
+        case FunctionCall:
+        case FunctionResponse:
+          throw new AIJSXError(
+            'Anthropic models do not support functions.',
+            ErrorCode.AnthropicDoesNotSupportFunctions,
+            'user'
+          );
+        default:
+          throw new AIJSXError(
+            `ChatCompletion's prompts must be UserMessage or AssistantMessage, but this child was ${message.tag.name}`,
+            ErrorCode.ChatCompletionUnexpectedChild,
+            'internal'
+          );
+      }
+    })
+  );
+
+  if (!messages.length) {
+    throw new AIJSXError(
+      "ChatCompletion must have at least child that's a SystemMessage, UserMessage, AssistantMessage, FunctionCall, or FunctionResponse, but no such children were found.",
```

Comment:
"one child"? Also, this is kind of confusing when the model doesn't support some of those types.